sormas-backend/src/main/java/de/symeda/sormas/backend/customizableenum/CustomizableEnumFacadeEjb.java

```diff
@@ -169,46 +159,71 @@ public boolean hasEnumValues(CustomizableEnumType type, Disease disease) {
 		return !getEnumValues(type, disease).isEmpty();
 	}
 
+	private <T extends CustomizableEnum> T buildCustomizableEnum(CustomizableEnumType type, String value, Language language, Class<T> enumClass) {
+		try {
+			T enumValue = enumClass.getDeclaredConstructor().newInstance();
+			enumValue.setValue(value);
+			enumValue.setCaption(enumValuesByLanguage.get(enumClass).get(language).get(value));
+			enumValue.setProperties(enumProperties.get(type).get(value));
+			return enumValue;
+		} catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
 	private synchronized <T extends CustomizableEnum> void fillLanguageCache(CustomizableEnumType type, Class<T> enumClass, Language language) {
 
-		if (!enumValuesByLanguage.get(enumClass).containsKey(language)) {
-			Map<String, String> enumValues = new HashMap<>();
-			for (CustomizableEnumValue customizableEnumValue : enumValueEntities.get(type)) {
-				if (StringUtils.equals(configFacade.getCountryLocale(), language.getLocale().toString())
-					|| CollectionUtils.isEmpty(customizableEnumValue.getTranslations())) {
-					// If the enum value does not have any translations or the user uses the server language,
-					// add the server language to the cache and use the default caption of the enum value
-					enumValues.putIfAbsent(customizableEnumValue.getValue(), customizableEnumValue.getCaption());
+		// may have already been put by another thread while this one was waiting (synchronized)
+		if (enumValuesByLanguage.get(enumClass).containsKey(language)) {
+			return;
+		}
+
+		boolean isCountryLanguage = StringUtils.equals(configFacade.getCountryLocale(), language.getLocale().toString());
+
+		Map<String, String> languageEnumValues = new HashMap<>();
+		for (CustomizableEnumValue customizableEnumValue : enumValueEntities.get(type)) {
```

Comment:
It might be a bit cleaner to stream enumValueEntities then collected in a map

Reply:
I'd leave it as-is.